sdk/python/feast/client.py

```diff
@@ -883,39 +861,27 @@ def _build_feature_references(
 
     for feature_ref in feature_refs:
         project_split = feature_ref.split("/")
-        version = 0
 
         if len(project_split) == 2:
-            project, feature_version = project_split
+            project, name = project_split
         elif len(project_split) == 1:
-            feature_version = project_split[0]
+            name = project_split[0]
             if default_project is None:
                 raise ValueError(
                     f"No project specified in {feature_ref} and no default project provided"
                 )
             project = default_project
         else:
             raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature:version"'
-            )
-
-        feature_split = feature_version.split(":")
-        if len(feature_split) == 2:
-            name, version = feature_split
-            version = int(version)
-        elif len(feature_split) == 1:
-            name = feature_split[0]
-        else:
-            raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature:version"'
+                f'Could not parse feature ref {feature_ref}, expecting "project/feature"'
             )
 
-        if len(project) == 0 or len(name) == 0 or version < 0:
+        if len(project) == 0 or len(name) == 0:
             raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature:version"'
+                f'Could not parse feature ref {feature_ref}, expecting "project/feature"'
             )
 
-        features.append(FeatureReference(project=project, name=name, version=version))
+        features.append(FeatureReference(project=project, name=name.split(":")[0]))
```

Comment:
Would it be better to just allow the name to be sent as is and give the user an exception?

If people are bumping their SDK version then we can encourage them to use the new feature references. I think we should just ensure that old clients are still supported by the serving layer, which you have already done.

Reply:
cc @mrzzy

Reply:
In this case, since versions are completely removed from the model and `fs:1` & `fs:2` no longer refer to different feature sets I think we should make it clear cut in SDKs (ie make the breaking chnage) that versions are no longer supported, otherwise we mind run into queries about why updating `fs:2` into `fs:3` broke some part of their code that depended on `fs:1` etc.

Reply:
I think for the user 0.5 should only phase out versions in a soft way, throwing deprecation warnings if versions are provided, and only start throwing exceptions in subsequent versions.

Reply:
Is there a specific scenario that the above code would help? It seems like we are already doing this on the server side. The upgrade path we are planning is to bump the server (because clients are more likely to be out of date). Then notify all clients to upgrade to the latest version.

0.4 client to 0.4 server works and 0.4 client to 0.5 server works (ignores version). So it seems like what you are optimizing for is a 0.5 client to a 0.4 server?

Reply:
Fair point, I've removed the code stripping versions from the sdks

Reply:
I've created an issue to discuss this in more detail #687. We should have a consistent policy for API compatibility across all PRs moving forward.